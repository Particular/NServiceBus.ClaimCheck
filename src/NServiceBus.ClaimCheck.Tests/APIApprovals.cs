namespace NServiceBus.ClaimCheck.Tests;

using Features;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void ApproveClaimCheck()
    {
        var publicApi = typeof(ClaimCheckFeature).Assembly.GeneratePublicApi(new ApiGeneratorOptions
        {
            ExcludeAttributes = ["System.Runtime.Versioning.TargetFrameworkAttribute", "System.Reflection.AssemblyMetadataAttribute"]
        });
        Approver.Verify(publicApi);
    }
}