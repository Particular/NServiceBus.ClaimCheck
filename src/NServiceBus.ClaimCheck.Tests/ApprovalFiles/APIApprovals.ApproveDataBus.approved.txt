[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.ClaimCheck.Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
namespace NServiceBus.ClaimCheck
{
    public class ClaimCheckConventions
    {
        public ClaimCheckConventions() { }
        public bool IsClaimCheckProperty(System.Reflection.PropertyInfo property) { }
    }
    public abstract class ClaimCheckDefinition
    {
        protected ClaimCheckDefinition() { }
        protected abstract System.Type ProvidedByFeature();
    }
    public class ClaimCheckExtensions : NServiceBus.Configuration.AdvancedExtensibility.ExposeSettings
    {
        public ClaimCheckExtensions(NServiceBus.Settings.SettingsHolder settings) { }
        public NServiceBus.ClaimCheck.ClaimCheckExtensions AddDeserializer<TSerializer>()
            where TSerializer : NServiceBus.ClaimCheck.IClaimCheckSerializer, new () { }
        public NServiceBus.ClaimCheck.ClaimCheckExtensions AddDeserializer<TSerializer>(TSerializer serializer)
            where TSerializer : NServiceBus.ClaimCheck.IClaimCheckSerializer { }
    }
    public class ClaimCheckExtensions<T> : NServiceBus.ClaimCheck.ClaimCheckExtensions
        where T : NServiceBus.ClaimCheck.ClaimCheckDefinition
    {
        public ClaimCheckExtensions(NServiceBus.Settings.SettingsHolder settings) { }
    }
    public static class ClaimCheckHeaders
    {
        public const string DataBusConfigContentType = "NServiceBus.DataBusConfig.ContentType";
    }
    public class ClaimCheckProperty<T> : NServiceBus.ClaimCheck.IClaimCheckProperty
        where T :  class
    {
        public ClaimCheckProperty() { }
        public ClaimCheckProperty(T value) { }
        public bool HasValue { get; set; }
        public string Key { get; set; }
        [System.Text.Json.Serialization.JsonIgnore]
        public System.Type Type { get; }
        [System.Text.Json.Serialization.JsonIgnore]
        [System.Xml.Serialization.XmlIgnore]
        public T Value { get; }
        public object GetValue() { }
        public void SetValue(object valueToSet) { }
    }
    public static class ConfigureFileShareClaimCheck
    {
        public static NServiceBus.ClaimCheck.ClaimCheckExtensions<NServiceBus.ClaimCheck.FileShareClaimCheck> BasePath(this NServiceBus.ClaimCheck.ClaimCheckExtensions<NServiceBus.ClaimCheck.FileShareClaimCheck> config, string basePath) { }
    }
    public static class ConventionsBuilderExtensions
    {
        public static NServiceBus.ConventionsBuilder DefiningDataBusPropertiesAs(this NServiceBus.ConventionsBuilder builder, System.Func<System.Reflection.PropertyInfo, bool> definesDataBusProperty) { }
    }
    public class FileShareClaimCheck : NServiceBus.ClaimCheck.ClaimCheckDefinition
    {
        public FileShareClaimCheck() { }
        protected override System.Type ProvidedByFeature() { }
    }
    public interface IClaimCheck
    {
        System.Threading.Tasks.Task<System.IO.Stream> Get(string key, System.Threading.CancellationToken cancellationToken = default);
        System.Threading.Tasks.Task<string> Put(System.IO.Stream stream, System.TimeSpan timeToBeReceived, System.Threading.CancellationToken cancellationToken = default);
        System.Threading.Tasks.Task Start(System.Threading.CancellationToken cancellationToken = default);
    }
    public interface IClaimCheckProperty
    {
        bool HasValue { get; set; }
        string Key { get; set; }
        System.Type Type { get; }
        object GetValue();
        void SetValue(object value);
    }
    public interface IClaimCheckSerializer
    {
        string ContentType { get; }
        object Deserialize(System.Type propertyType, System.IO.Stream stream);
        void Serialize(object databusProperty, System.IO.Stream stream);
    }
    public class SystemJsonClaimCheckSerializer : NServiceBus.ClaimCheck.IClaimCheckSerializer
    {
        public SystemJsonClaimCheckSerializer() { }
        public string ContentType { get; }
        public object Deserialize(System.Type propertyType, System.IO.Stream stream) { }
        public void Serialize(object dataBusProperty, System.IO.Stream stream) { }
    }
}
namespace NServiceBus.Features
{
    public class ClaimCheckFeature : NServiceBus.Features.Feature
    {
        protected override void Setup(NServiceBus.Features.FeatureConfigurationContext context) { }
    }
}
namespace NServiceBus
{
    public static class UseClaimCheckExtensions
    {
        public static NServiceBus.ClaimCheck.ClaimCheckExtensions UseClaimCheck(this NServiceBus.EndpointConfiguration config, System.Func<System.IServiceProvider, NServiceBus.ClaimCheck.IClaimCheck> dataBusFactory, NServiceBus.ClaimCheck.IClaimCheckSerializer dataBusSerializer) { }
        public static NServiceBus.ClaimCheck.ClaimCheckExtensions<TDataBusDefinition> UseClaimCheck<TDataBusDefinition>(this NServiceBus.EndpointConfiguration config, NServiceBus.ClaimCheck.IClaimCheckSerializer dataBusSerializer)
            where TDataBusDefinition : NServiceBus.ClaimCheck.ClaimCheckDefinition, new () { }
        public static NServiceBus.ClaimCheck.ClaimCheckExtensions<TDataBusDefinition> UseClaimCheck<TDataBusDefinition, TDataBusSerializer>(this NServiceBus.EndpointConfiguration config)
            where TDataBusDefinition : NServiceBus.ClaimCheck.ClaimCheckDefinition, new ()
            where TDataBusSerializer : NServiceBus.ClaimCheck.IClaimCheckSerializer, new () { }
    }
}